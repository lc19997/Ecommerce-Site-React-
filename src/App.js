import { Preloader } from 'components/common';
import React, { StrictMode } from 'react';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';
import AppRouter from 'routers/AppRouter';

const App = ({ store, persistor }) => (
	<StrictMode>
		<Provider store={store}>
			<PersistGate loading={<Preloader />} persistor={persistor}>
				<AppRouter />
			</PersistGate>
		</Provider>
	</StrictMode>
);

export default App;
