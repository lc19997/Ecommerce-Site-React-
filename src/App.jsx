/* eslint-disable react/forbid-prop-types */
import { Preloader } from 'components/common';
import PropType from 'prop-types';
import React, { StrictMode } from 'react';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';
import AppRouter from 'routers/AppRouter';

const App = ({ store, persistor }) => (
  <StrictMode>
    <Provider store={store}>
      <PersistGate loading={<Preloader />} persistor={persistor}>
        <AppRouter />
      </PersistGate>
    </Provider>
  </StrictMode>
);

App.propTypes = {
  store: PropType.any.isRequired,
  persistor: PropType.any.isRequired
};

export default App;
