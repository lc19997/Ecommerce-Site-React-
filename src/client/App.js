import React from 'react';
import AppRouter from './routers/AppRouter';

const App = () => {
  return (
    <AppRouter />
  );
};

export default App;
