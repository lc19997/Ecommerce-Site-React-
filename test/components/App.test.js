import React from 'react';
import { shallow } from 'enzyme';
import App from '../../src/client/components/App';

test('Should render App component', () => {
  const wrapper = shallow(<App />);

  expect(wrapper).toMatchSnapshot();
});

// More tests ...
