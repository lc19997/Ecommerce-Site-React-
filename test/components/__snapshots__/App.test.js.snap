// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render App component 1`] = `
<div>
  <h1>
    React Webpack Boilerplate
  </h1>
  <p>
    Edit App component in src folder to get started.
  </p>
</div>
`;
